src/worldcereal/rdm_api/rdm_interaction.py

```diff
@@ -169,12 +211,104 @@ def _get_download_urls(
+
+        Parameters
+        ----------
+        ewoc_codes: Optional[List[int]] = None
+            A list of EWOC codes for which crop counts need to be fetched.
+        collection_ids : Optional(List[str]), optional
+            List of collection IDs to check.
+            If not specified, all collections containing the crop of interest will be checked,
+            which match any of the criteria specified in the other parameters.
+        geometry : Optional[BaseGeometry], optional
+            A user-defined geometry for which all intersecting collections need to be found.
+            CRS should be EPSG:4326.
+            If None, all available data will be queried. By default None.
+        temporal_extent : Optional[List[str]], optional
+            A list of two strings representing the temporal extent, by default None.
+            If None, all available data will be queried.
+        include_public: Optional[bool] = True
+            Whether or not to include public collections.
+        include_private: Optional[bool] = False
+            Whether or not to include private collections.
+
+        Returns
+        -------
+        Pandas DataFrame
+            Dataframe containing for each collection id the number of items matching the requested crop types.
+            If stats are not available, the count will be -9999.
+        """
+
+        # If no collection IDs are provided,
+        # get all collections containing the crop types of interest
+        if not collection_ids:
+            collections = self.get_collections(
+                geometry=geometry,
+                temporal_extent=temporal_extent,
+                ewoc_codes=ewoc_codes,
+                include_private=include_private,
+                include_public=include_public,
+            )
+            if not collections:
+                return pd.DataFrame()
+            collection_ids = [col.id for col in collections]
+
+        logger.info(f"Querying {len(collection_ids)} collections...")
+
+        # Prepare result
+        result = []
+
+        for col_id in collection_ids:
+            itemUrl = f"{self.RDM_ENDPOINT}/collections/{col_id}/items/codestats"
+            itemsResponse = requests.get(itemUrl)
+            res = itemsResponse.json()
+            if "ewocStats" in res:
+                stats = res["ewocStats"]
+                for cropcode in ewoc_codes:
+                    count = 0
+                    for stat in stats:
+                        if stat["code"] == cropcode:
+                            count = stat["count"]
+                            break
+                    result.append(
+                        {"collectionId": col_id, "EwocCode": cropcode, "count": count}
+                    )
+            else:
+                # statistics not available
+                for cropcode in ewoc_codes:
+                    result.append(
+                        {"collectionId": col_id, "EwocCode": cropcode, "count": -9999}
```

Comment:
why -9999 instead of NaN?

Reply:
I would like to make a distinction between:
- crop statistics not available for a collection
- crop statistics available, but crop is absent from the collection.
Hence the use of -9999

Reply:
fixed it by issuing a warning instead of showing the collection with missing crop counts